homeassistant/components/recorder/purge.py

```diff
@@ -112,6 +115,223 @@ def _select_event_state_and_attributes_ids_to_purge(
     return event_ids, state_ids, attributes_ids
 
 
+def _state_attrs_exist(attr: int | None) -> Select:
+    """Check if a state attributes id exists in the states table."""
+    return select(func.min(States.attributes_id)).where(States.attributes_id == attr)
+
+
+def _generate_find_attr_lambda(
```

Comment:
The docs for sqlalchemy say:
```
The best way to resolve the above situation is to not refer to foo inside of the lambda, and refer to it outside instead:

>>> def my_stmt(foo):
...     x_param, y_param = foo.x, foo.y
...     stmt = lambda_stmt(lambda: select(func.max(x_param, y_param)))
...     return stmt
```

So even if I pass this as a list I'm stuck unpacking them as `attr1, attr2, attr3, attr4, .... = attrs` outside the lambda